### ISSUE TITLE
Bogus undefined name error with generic type alias

### ISSUE THREAD

**Issue by Oberon00** [python/mypy#7092]
* Are you reporting a bug, or opening a feature request?

Bug.

* Please insert below the code you are checking with mypy,
  or a mock-up repro if the source is private. We would appreciate
  if you try to simplify your case to a minimal repro.

This requires two modules to trigger the bug:
dummy.py:
```py
from typing import Type, TypeVar, Callable

T = TypeVar('T')
ImplementationFactory = Callable[[Type[T]], T]

MyFactory = ImplementationFactory[str] # Works fine
```

dummy2.py:
```py
import dummy

MyFactory2 = dummy.ImplementationFactory[str]
```

* What is the actual behavior/output?
```
$ mypy dummy2.py
dummy2.py:4: error: Name 'T' is not defined
```

Note that line 4 is empty. You can also add comments to get a different line number displacement. Also note that `T` is never referred to in `dummy2`.

* What is the behavior/output you expect?

I would expect no errors for this code.

* What are the versions of mypy and Python you are using?

```
$ mypy --version
mypy 0.711
$ python --version
Python 3.7.3
```

*  Do you see the same issue after installing mypy from Git master?

I didn't try.

* What are the mypy flags you are using? (For example --strict-optional)

None. The command line is shown in full, there is no config file.

**Comment by JelleZijlstra** [comment]
This sounds similar to #7077.

**Comment by Oberon00** [comment]
Maybe similar, but this happens also without any special configuration so I'd say it is more severe based on that.

Can this issue get any labels, like "bug"?

**Comment by ilevkivskyi** [comment]
This still happens on master (and is pretty embarrassing).

**Comment by Oberon00** [comment]
Any news on this? This is currently causing us to copy & paste type definitions around in https://github.com/open-telemetry/opentelemetry-python (e.g. https://github.com/open-telemetry/opentelemetry-python/blob/master/opentelemetry-api/src/opentelemetry/trace/__init__.py#L497-L502)

**Comment by pappasam** [comment]
I've tried this example on mypy version 0.790 and am not receiving any error. Maybe this issue is fixed?

**Comment by hauntsaninja** [comment]
Thanks for checking! mypy_primer indicates this was fixed by #7885
